swatch-metrics-hbi/ct/java/com/redhat/swatch/hbi/events/ct/api/MessageValidators.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.hbi.events.ct.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.swatch.component.tests.api.MessageValidator;
+import com.redhat.swatch.component.tests.utils.JsonUtils;
+import com.redhat.swatch.hbi.events.dtos.hbi.HbiEvent;
+import com.redhat.swatch.hbi.events.dtos.hbi.HbiHostCreateUpdateEvent;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+import org.candlepin.subscriptions.json.Event;
+
+public class MessageValidators {
+  public static MessageValidator<HbiEvent> eventMatches(String type) {
+    return new MessageValidator<>(event -> type.equals(event.getType()), HbiEvent.class);
+  }
+
+  public static MessageValidator<HbiEvent> eventMatches(String type, String requestId) {
+    return new MessageValidator<>(
+        event ->
+            type.equals(event.getType()) && requestId.equals(event.getMetadata().getRequestId()),
+        HbiEvent.class);
+  }
+
+  public static MessageValidator<HbiEvent> hbiEventEquals(HbiHostCreateUpdateEvent hbiEvent) {
+    return new MessageValidator<>(
+        event -> {
+          try {
+            var mapper = JsonUtils.getObjectMapper();
+            var expected = mapper.valueToTree(hbiEvent);
+            var actual = mapper.valueToTree(event);
+            return expected.equals(actual);
```

Comment:
Why we can't we just use equals in these validators? What was driving the need for the mapper?